clientapi/routing/register.go

```diff
@@ -935,7 +958,7 @@ func RegisterAvailable(
 	}
 }
 
-func handleSharedSecretRegistration(userAPI userapi.UserInternalAPI, sr *SharedSecretRegistration, req *http.Request) util.JSONResponse {
+func handleSharedSecretRegistration(userAPI userapi.UserInternalAPI, sr *SharedSecretRegistration, passwordRequirements config.PasswordRequirements, req *http.Request) util.JSONResponse {
```

Comment:
I'm a little concerned here. I'm not really sure if this shared secret registration should share the same requirements as user passwords. I'm happy to have it do the defaults instead (8 minlength & 512minlength and nothing else) regardless of what the user specifies in the config.

Reply:
I think it's fine to use the same requirements, as shared secret registration isn't any different to a normal registration.

Reply:
Agreed, let's have a single consistent ruleset for passwords.